src/Symfony/Component/Filesystem/Tests/PathTest.php

```diff
@@ -0,0 +1,1341 @@
+            array('/webmozart/puli', false, ''),
+            array('/', false, ''),
+            array('', false, ''),
+
+            array('/webmozart/puli/style.CSS', false, 'CSS'),
+            array('/webmozart/puli/style.CSS', true, 'css'),
+            array('/webmozart/puli/style.ÄÖÜ', false, 'ÄÖÜ'),
+        );
+
+        if (extension_loaded('mbstring')) {
+            // This can only be tested, when mbstring is installed
+            $tests[] = array('/webmozart/puli/style.ÄÖÜ', true, 'äöü');
+        }
+
+        return $tests;
+    }
+
+    /**
+     * @dataProvider provideGetExtensionTests
+     */
+    public function testGetExtension($path, $forceLowerCase, $extension)
+    {
+        $this->assertSame($extension, Path::getExtension($path, $forceLowerCase));
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The path must be a string. Got: array
+     */
+    public function testGetExtensionFailsIfInvalidPath()
+    {
+        Path::getExtension(array());
+    }
+
+    public function provideHasExtensionTests()
+    {
+        $tests = array(
+            array(true, '/webmozart/puli/style.css.twig', null, false),
+            array(true, '/webmozart/puli/style.css', null, false),
+            array(false, '/webmozart/puli/style.css.', null, false),
+            array(false, '/webmozart/puli/', null, false),
+            array(false, '/webmozart/puli', null, false),
+            array(false, '/', null, false),
+            array(false, '', null, false),
+
+            array(true, '/webmozart/puli/style.css.twig', 'twig', false),
+            array(false, '/webmozart/puli/style.css.twig', 'css', false),
+            array(true, '/webmozart/puli/style.css', 'css', false),
+            array(true, '/webmozart/puli/style.css', '.css', false),
+            array(true, '/webmozart/puli/style.css.', '', false),
+            array(false, '/webmozart/puli/', 'ext', false),
+            array(false, '/webmozart/puli', 'ext', false),
+            array(false, '/', 'ext', false),
+            array(false, '', 'ext', false),
+
+            array(false, '/webmozart/puli/style.css', 'CSS', false),
+            array(true, '/webmozart/puli/style.css', 'CSS', true),
+            array(false, '/webmozart/puli/style.CSS', 'css', false),
+            array(true, '/webmozart/puli/style.CSS', 'css', true),
+            array(true, '/webmozart/puli/style.ÄÖÜ', 'ÄÖÜ', false),
+
+            array(true, '/webmozart/puli/style.css', array('ext', 'css'), false),
+            array(true, '/webmozart/puli/style.css', array('.ext', '.css'), false),
+            array(true, '/webmozart/puli/style.css.', array('ext', ''), false),
+            array(false, '/webmozart/puli/style.css', array('foo', 'bar', ''), false),
+            array(false, '/webmozart/puli/style.css', array('.foo', '.bar', ''), false),
+        );
+
+        if (extension_loaded('mbstring')) {
```

Comment:
The component already makes use of the mbstring polyfill, so we should not need to check for the extension.